src/black/parsing.py

```diff
@@ -108,26 +108,34 @@ def lib2to3_unparse(node: Node) -> str:
 
 def parse_ast(src: str) -> Union[ast.AST, ast3.AST, ast27.AST]:
     filename = "<unknown>"
+    error = None
     if sys.version_info >= (3, 8):
         # TODO: support Python 4+ ;)
         for minor_version in range(sys.version_info[1], 4, -1):
             try:
                 return ast.parse(src, filename, feature_version=(3, minor_version))
-            except SyntaxError:
+            except SyntaxError as e:
+                if error is None:
+                    error = str(e)
                 continue
     else:
         for feature_version in (7, 6):
             try:
                 return ast3.parse(src, filename, feature_version=feature_version)
-            except SyntaxError:
+            except SyntaxError as e:
+                if error is None:
+                    error = str(e)
                 continue
     if ast27.__name__ == "ast":
         raise SyntaxError(
             "The requested source code has invalid Python 3 syntax.\n"
             "If you are trying to format Python 2 files please reinstall Black"
             " with the 'python2' extra: `python3 -m pip install black[python2]`."
         )
-    return ast27.parse(src)
+    try:
+        return ast27.parse(src)
+    except SyntaxError as e:
+        raise SyntaxError(error or str(e))
```

Comment:
Firstly, thanks for the PR!

If I'm not mistaken, this means that 2.7 errors are never passed onwards. To get to this `try` some other `try` above must fail, then `error` always has a value and the `or` short circuits. Indeed this only ever prints the error for Python 3.6, because it's the last to fail before 2.7. So this moves the problem from 2.7 to 3.6, but I'm pretty sure this is not quite the optimal solution.

The linked issue spoke of reparsing with the latest Python version available, and that idea had some agreement. That could be possible using `sys.version_info` and iffing the parser versions again.